java-backend/src/main/java/org/kframework/backend/java/kil/TermContext.java

```diff
@@ -17,38 +15,32 @@
 /**
  * An object containing context specific to a particular configuration.
  */
-public class TermContext extends JavaSymbolicObject {
+public class TermContext {
 
     private final AtomicLong counter;
 
     private final GlobalContext global;
 
-    private Set<Variable> initialVariables;
+    private Set<Variable> initialLhsVariables;
 
     private Term topTerm;
 
     private ConjunctiveFormula topConstraint;
 
-    private KOREtoBackendKIL converter;
```

Comment:
i cannot see why this is necessarily safe.

Reply:
Same as above. If it wouldn't be safe, code wouldn't compile.